### DISCUSSION TITLE
choose |x| P(x) should require exists |x| P(x)

### DISCUSSION THREAD

**Discussion by jonhnet** [verus-lang/verus#802] [Language design]
We want this for the same reason that eager requires checking is desirable in other uses of spec functions: It is frustrating to take a half dozen debugging steps only to discover that the thing we reasonably assumed was true when we called choose wasn't actually. I want that error eagerly.

**Comment by Chris-Hawblitzel** [comment]
See https://github.com/verus-lang/verus/issues/408

**Comment by utaal** [comment]
This is going to be addressed by fixes for #408 and by-default "recommends" checking as proposed in the (WIP) https://github.com/verus-lang/verus/discussions/764, so closing for now.
